tests/platform_helper/utils/test_validation.py

```diff
@@ -596,6 +583,44 @@ def test_load_and_validate_config_valid_file(yaml_file):
     assert validated == conf
 
 
+def test_validation_fails_if_invalid_default_version_keys_present(
+    fakefs, capsys, valid_platform_config
+):
+    valid_platform_config["default_versions"] = {"something-invalid": "1.2.3"}
+    Path(PLATFORM_CONFIG_FILE).write_text(yaml.dump(valid_platform_config))
+
+    with pytest.raises(SchemaError) as ex:
+        load_and_validate_platform_config()
+
+    assert "Wrong key 'something-invalid'" in str(ex)
+
+
+def test_validation_fails_if_invalid_environment_version_override_keys_present(
+    fakefs, capsys, valid_platform_config
+):
+    valid_platform_config["environments"]["*"]["versions"] = {"platform-helper": "1.2.3"}
```

Comment:
I'm a bit confused by this test.  Is 'platform-helper' an invalid key?  The test above is more obvious, with invalid key 'something-invalid'.  It's all made more confusing since the variable being used is called valid_platform_config - but isn't the point of the test that the config is invalid?

Reply:
Yes - platform-helper version cannot be overridden at an environment level. In the pipelines currently we install the required version of platform-helper in the build stage and all environments use that same version.

Reply:
I'll rename the test.